File: app/src/main/java/org/senatov/compare/DirCompareResult.java
/*
 * DirCompareResult.java — result of tree-based directory comparison.
 * Holds paired DirTreeModels and diff count.
 * Iakov Senatov, 2026
 */
package org.senatov.compare;

import org.senatov.model.tree.DirTreeModel;


public record DirCompareResult(
        DirTreeModel leftModel,
        DirTreeModel rightModel,
        int diffCount
) {

    public boolean isIdentical() {
        return diffCount == 0;
    }


    public String statusText() {
        if (isIdentical()) {
            return "✅ dirs identical";
        }
        return "≠ " + diffCount + " differences";
    }
}

File: app/src/main/kotlin/org/senatov/compare/DirCompareResult.kt
/*
 * DirCompareResult — result of tree-based dir comparison.
 * paired DirTreeModels + diff count.
 * Iakov Senatov, 2026
 */
package org.senatov.mimicomparator.compare

import org.senatov.mimicomparator.model.tree.DirTreeModel


data class DirCompareResult(
    val leftModel: DirTreeModel,
    val rightModel: DirTreeModel,
    val diffCount: Int
) {

    val isIdentical: Boolean get() = diffCount == 0


    fun statusText(): String =
        if (isIdentical) "✅ dirs identical"
        else "≠ $diffCount differences"
}